client/src/Login.tsx

```diff
@@ -1,63 +1,147 @@
+  let { state } = useLocation();
+  const autoLoad = state === null ? true : false;
+
   const [accessKey, setAccessKey] = useState("");
   const [secretKey, setSecretKey] = useState("");
   const [clusterName, setClusterName] = useState("");
   const [regionName, setRegionName] = useState("");
+  // Toggling Password visiblity
+  const [hideEye, toggleEye] = useState(true);
+  const [hidePassword, togglePassword] = useState(true);
+  // Whether to show the loading screen
+  const [isLoading, setLoading] = useState(false);
+
+  // Only fire on initial render
+  // Executes twice because of React.Strict
+  useEffect(() => {
+    // On mount
+    // Don't load user in if they just logged out
+    if(autoLoad !== false) {
+      loginUser(true);
+    } else {
+      setLoading(true);
+      // Then load config
+      window.electronAPI.getConfig();
+    }
+    // Response after trying to update local cred/config files
+    window.electronAPI.onConfigResp('onConfigResp', (event: any, data: any) => {
+      console.log("From server:", data)
+      setLoading(false);
+      // The fourth datapoint is whether the kube config was set properly
+      // Only attempt to log the user in if kube config was created
+      if(data[4]) {
+        loginUser(false);
+      } else {
+        alert(badCredentials);
+      }
+    })
+    // Why is this executing twice?
+    // Response after parsing local cred/config files
+    window.electronAPI.onSendConfig('onSendConfig', (event: any, data: [string, string, string, string]) => {
+      console.log("Parsed from local files:", data);
+      setLoading(false);
+      // Update each input field
+      setAccessKey(data[0]);
+      setSecretKey(data[1]);
+      setClusterName(data[2]);
+      setRegionName(data[3]);
+    })
 
-  // If the server sends anything besides 200, then the user isn't authenticated
-  // For the user's convenience, find and parse their three files (in $HOME/.aws/ 
-  // and $HOME/.kube/) and populate their corresponding fields in the form
-  const parseLocalCredentials = () => {
-    console.log('<ELECTRON NEEDED> : Parse User\'s local files here!');
-  }
+    return () => {
+      console.log('useEffect: unmount');
+      window.electronAPI.unMount();
+    };
+  }, [])
 
-  const loginUser = () => {
+  const loginUser = (loadConfig : boolean) => {
     //alert('id:' + accessKey + ' secret:' + secretKey);
 
     fetch('http://localhost:3000/authenticate', {
       method: 'GET'
     })
-    .then(response => {
-      //console.log('resp:', resp.json());
-      //console.log('typeof:', typeof response.json());
-      //console.log(response.json());
-      console.log('status:', response.status);
-      if(response.status !== 200) {
-        console.log('bad credentials...');
+      .then(response => {
+        //console.log('resp:', resp.json());
+        //console.log('typeof:', typeof response.json());
+        //console.log(response.json());
```

Comment:
Should these comment logs remain here? Are each of them intended to be of use for future development? If so, leave a comment describe the utility of the logs for other devs.

Reply:
I haven't fully tested the behavior in all cases. For example, what would be the response if the user doesn't have access to the internet?